easybuild/easyblocks/s/scipy.py

```diff
@@ -30,34 +30,179 @@
-from easybuild.easyblocks.generic.pythonpackage import det_pylibdir
+import easybuild.tools.environment as env
 import easybuild.tools.toolchain as toolchain
+from easybuild.easyblocks.generic.fortranpythonpackage import FortranPythonPackage
+from easybuild.easyblocks.generic.mesonninja import MesonNinja
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage, det_pylibdir
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import change_dir, copy_dir
 
 
-class EB_scipy(FortranPythonPackage):
+class EB_scipy(FortranPythonPackage, PythonPackage, MesonNinja):
     """Support for installing the scipy Python package as part of a Python installation."""
 
+    @staticmethod
+    def extra_options():
+        """Easyconfig parameters specific to scipy."""
+        extra_vars = ({
+            'enable_slow_tests': [False, "Run scipy test suite, including tests marked as slow", CUSTOM],
+            # ignore test failures by default in scipy < 1.9 to maintain behaviour in older easyconfigs
+            'ignore_test_result': [None, "Run scipy test suite, but ignore test failures (True/False/None). Default "
+                                         "(None) implies True for scipy < 1.9, and False for scipy >= 1.9", CUSTOM],
+        })
+
+        return PythonPackage.extra_options(extra_vars=extra_vars)
+
     def __init__(self, *args, **kwargs):
         """Set scipy-specific test command."""
-        super(EB_scipy, self).__init__(*args, **kwargs)
-
+        # calling PythonPackage __init__ also lets MesonNinja work in an extension
+        PythonPackage.__init__(self, *args, **kwargs)
         self.testinstall = True
-        self.testcmd = "cd .. && %(python)s -c 'import numpy; import scipy; scipy.test(verbose=2)'"
+
+        if LooseVersion(self.version) >= LooseVersion('1.9'):
+            self.use_meson = True
+
+            # enforce scipy test suite results if not explicitly disabled for scipy >= 1.9
+            # strip inherited PythonPackage installopts
+            installopts = self.cfg['installopts']
+            pythonpackage_installopts = ['--no-deps', '--ignore-installed', '--no-index', '--egg',
+                                         '--zip-ok', '--no-index']
+            self.log.info("Stripping inherited PythonPackage installopts %s from installopts %s",
+                          pythonpackage_installopts, installopts)
+            for i in pythonpackage_installopts:
+                installopts = installopts.replace(i, '')
+            self.cfg['installopts'] = installopts
+
+        else:
+            self.use_meson = False
+
+        if self.cfg['ignore_test_result'] is None:
+            # enforce scipy test suite results if not explicitly disabled for scipy >= 1.9
+            if self.use_meson:
+                self.cfg['ignore_test_result'] = False
+            else:
+                self.cfg['ignore_test_result'] = True
```

Comment:
@jfgrimm Let's simplify this a bit, see https://github.com/jfgrimm/easybuild-easyblocks/pull/2